sdk/typescript/src/types/framework.ts

```diff
@@ -279,40 +270,19 @@ export class Coin {
           `${amountToSend}. Try reducing the transfer amount to ${suggestedAmountToSend}.`
       );
     }
-    if (!isSuiTransfer) {
-      const allGasCoins = allCoins.filter((aCoin) => Coin.isSUI(aCoin));
-      const gasCoin = Coin.selectCoinWithBalanceGreaterThanOrEqual(
-        allGasCoins,
-        BigInt(gasBudget)
-      );
-      if (!gasCoin) {
-        // TODO: denomination for gasBudget?
-        throw new Error(
-          `Unable to find a coin to cover the gas budget ${gasBudget}`
-        );
-      }
-    }
-    const signerAddress = await signer.getAddress();
-    const inputCoins = inputCoinObjs.map(Coin.getID);
-    const txCommon = {
-      inputCoins,
-      recipients: [recipient],
-      // TODO: change this to string to avoid losing precision
-      amounts: [Number(amountToSend)],
-      gasBudget: Number(gasBudget),
-    };
     if (isSuiTransfer) {
-      return signer.serializer.serializeToBytes(signerAddress, {
-        kind: 'paySui',
-        data: { ...txCommon },
-      });
+      inputCoinObjs.unshift(gasCoin);
     }
-    return signer.serializer.serializeToBytes(signerAddress, {
-      kind: 'pay',
-      data: { ...txCommon },
-      // we know there is a gas coin with enough balance to cover
-      // the gas budget let rpc select one for us
-    });
+    return {
+      kind: isSuiTransfer ? 'paySui' : 'pay',
+      data: {
+        inputCoins: inputCoinObjs.map(Coin.getID),
+        recipients: [recipient],
+        // TODO: change this to string to avoid losing precision
```

Comment:
Does this not work as a string right now? Seems like a bad bug cc @666lcz @gegaowp

Reply:
Any number bigger than Number.MAX_SAFE_INTEGER is not accurately represented https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

<img width="222" alt="Screenshot 2022-12-09 at 23 18 34" src="https://user-images.githubusercontent.com/10210143/206811123-358ba0e2-8d1f-4eaa-af64-37cdfb509f63.png">

Reply:
We should use [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) to avoid those cases (and convert them to string when we serialise them to json?)

Reply:
@gegaowp Yeah as mentioned by Pavlos, this should be a string. All u64s in APIs should be represented as string, u32 is the largest number we support that can be represented as a number in JSON.